tests/lang-c/test.el

```diff
@@ -1,18 +1,57 @@
+(defun adjust-pattern (pattern)
+  "Convert the pattern got from the value returned from `citre-get-definitions' to xref like string."
+  (let* (
+	 ;; Trim the initial '/' and the last /;".
+	 (pattern (substring pattern 1 (- (length pattern) 3)))
+	 ;; Trim the initial ^.
+	 (pattern (if (string-prefix-p "^" pattern)
+		      (substring pattern 1)
+		    pattern))
+	 ;; Trim the last $.
+	 (pattern (if (string-suffix-p "$" pattern)
+		      (substring pattern 0 (- (length pattern) 1))
+		    pattern))
+	 ;; Trim the initial whitespces.
+	 (pattern (if (string-match "^[ \t]+\\([^ \t].*\\)$" pattern)
+		      (match-string 1 pattern)
+		    pattern))
+	 )
+    pattern))
+
 (defun defs-to-xref (defs)
   "Convert the value returned from `citre-get-definitions' to xref like string."
-  (with-temp-buffer
-    (mapc (lambda (def)
-	    (insert (gethash 'ext-abspath def))
-	    (insert ?\n)
-	    (insert (gethash 'line def))
-	    (insert ": ")
-	    (insert (format "(%s)\n" (gethash 'ext-kind-full def))))
-	  defs)
-    (replace-regexp-in-string
-     (concat "^" (regexp-quote (expand-file-name default-directory)))
-     ""
-     (buffer-substring-no-properties
-      (point-min) (point-max)))))
+  (let (last-abspath)
+    (with-temp-buffer
+      (mapc (lambda (def)
+	      (let ((abspath (gethash 'ext-abspath def)))
+		(unless (equal abspath last-abspath)
+		  (insert abspath)
+		  (insert ?\n)
+		  (setq last-abspath abspath)))
+	      (insert (gethash 'line def))
```

Comment:
Can we use `citre-make-tag-str` here? I think we could assemble a string in xref style with it, using these components: `'(location :no-path t :suffix ":")`, `'(annotation :prefix "(" :suffix ")")` and `'(content)`.

Reply:
Thank you. With citre-make-tag-str, we can remove adjust-pattern!